Support parsing `@font-face`

yisibl opened the pull request:
In the future, we may also add `@font-palette-values` and `@font-feature-values`.

Part of https://github.com/linebender/resvg/issues/541

yisibl committed:
Supports parsing `@font-face`

Part of https://github.com/linebender/resvg/issues/541

yisibl commented:
@LaurenzV PTAL.

The current implementation only targets `@font-face`; I'm unsure whether a generic at-rule parser is required.

LaurenzV reviewed (commented):
LGTM, maybe add a similar test like `font_face_02_mixed_with_rules` but where font-face is not the first rule?

I think for now we don't need general at-rule support, the font-face one is by far the most important one.

yisibl commented:
Thank you, I've added the new test `font_face_03_mixed_with_rules`.

yisibl committed:
fix: doc_auto_cfg error

error[E0557]: feature has been removed
  --> src/lib.rs:42:29
   |
42 | #![cfg_attr(docsrs, feature(doc_auto_cfg))]
   |                             ^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in 1.92.0; see <https://github.com/rust-lang/rust/pull/138907> for more information
   = note: merged into `doc_cfg`

error: Compilation failed, aborting rustdoc

LaurenzV commented:
Doc failure needs to be fixed and then it should be good to go.

LaurenzV commented:
Actually, let me just fix it with a separate PR.

LaurenzV commented:
Hmm, is it possible that you haven't rebased onto the newest main? I can't find the line.

yisibl commented:
> Doc failure needs to be fixed and then it should be good to go.

Fixed:
https://github.com/linebender/simplecss/pull/39/changes/4ba165df13130b8ea7b9b068aa935458ec73b379

LaurenzV commented:
Nevermind, _I_ was the one looking at an old version. 😅

LaurenzV commented:
Thanks!
